spec/features/profile_spec.rb

```diff
@@ -1,23 +1,32 @@
 require 'rails_helper'
 
-feature 'Edit profile' do
-  let(:user) { create(:user) }
+RSpec.feature 'Edit profile' do
   let(:features) { FeaturesSpecHelper.new }
 
-  scenario 'A user edits its profile information and password' do
+  background do
+    @old_password = user.encrypted_password
     features.sign_in(user)
 
     click_on 'See my profile'
 
     click_on 'Edit profile'
 
-    select 'Sergent', from: 'user_rank'
+    select 'Caporal', from: 'user_rank'
     fill_in 'user_password', with: 'new_password'
     fill_in 'user_password_confirmation', with: 'new_password'
     fill_in 'user_current_password', with: user.password
     click_button 'Update'
 
-    expect(page).to have_content 'Your account has been updated successfully.'
-    expect(User.find_by(email: user.email).rank).to eq('sgt')
+    @updated_user = User.find_by(email: user.email)
```

Comment:
Our update process could involve a model, a service and a controller + its view (+ presenter).
If model and service are unit tested, what is left to test is the controller and the presenter/view.

In this way, we don't have to test the persistence, because it's not the responsibility of nether the controller, nor the presenter/view.
We should only check here that the whole process goes well, and pieces are playing good together with the controller/UI parts.

If the page displays correctly, that's enough to assume the process is working.
Indeed, if we want to add a bit more of confidence, we can also check the persistence by checking that "Caporal" is displayed somewhere on the page.

Checking this is the most we have to do, or that would say our design should be reviewed. :)